src/main/java/com/algorand/algosdk/transaction/Transaction.java

```diff
@@ -461,6 +466,7 @@ private Transaction(
         if (note != null) this.note = note;
         if (genesisID != null) this.genesisID = genesisID;
         if (genesisHash != null) this.genesisHash = genesisHash;
+        if (lease != null) this.lease = lease;
```

Comment:
Do we want a length check here, as well, or is that redundant?

Reply:
Unfortunately, all the members are public, including lease. Which means, any code can set the lease value to anything and cannot be checked. 

Setting it properly though setLease, will check the value. 

Checking the assignment here is not necessary, because it is a private constructor, called from a private method, used for parsing external transactions, which is assumed to be well-formed and not to violate the lease size.